### PR TITLE
Prevent crash when target is unavailible

### PR DESCRIPTION
**Pull request by Akxe**
- Added 500 error
- Added UTF-8 encoding for error pages (for multi language support)

### CODE REVIEW

File: router.js [hunk 1]
```diff
@@ -34,16 +34,27 @@ var configFileEvent = function (curr, prev) {
 
 var do404 = function (res) {
 	res.writeHead(404);
+	res.header("Content-Type", "application/json; charset=utf-8");
```
Comment (Irrelon) [inline router.js]: Should we be providing an application/json header for a 404 with no JSON output? This seems wrong to me.

**Comment by Irrelon** [comment]
Thanks for this pull request :+1: that's really nice of you to take the time to contribute :)

**Comment by Akxe** [comment]
I use it so much, I thought I may give some love back ;)
I did some optimizations to the code, but it also changes how the input config is written from plain JS to somewhat "user friendly" kind... The main thing is, that I used a Map (from ES6) for routerTable and Route class (made), so that the V8 can do its optimizations... If you want I can post it

File: router.js (hunk 1 above)
Comment (Akxe) [inline router.js]: My bad... true, I just wanted the UTF-8 for non english languages...

**Comment by Irrelon** [comment]
@Akxe Sure thing, post a pull request and I'll take a look at your enhancements :+1: 

The config format is open to change anyway so if it makes things easier to understand then all the better.

**Pull request merged**